src/Native/Unix/System.Security.Cryptography.Native/pal_bio.cpp

```diff
@@ -52,3 +118,29 @@ extern "C" int32_t CryptoNative_BioCtrlPending(BIO* bio)
     assert(result <= INT32_MAX);
     return static_cast<int32_t>(result);
 }
+
+extern "C" void CryptoNative_BioSetAppData(BIO* bio, void* data)
+{
+    BIO_set_ex_data(bio, 0, data);
+}
+
+extern "C" void CryptoNative_BioSetWriteFlag(BIO* bio)
+{
+    BIO_set_flags(bio, BIO_FLAGS_WRITE);
+}
+
+extern "C" void CryptoNative_BioSetShoudRetryReadFlag(BIO* bio)
+{
+    BIO_set_flags(bio, BIO_FLAGS_SHOULD_RETRY | BIO_FLAGS_READ);
+}
```

Comment:
Well, the issue was we then needed a whole Enums matched to Enums deal in the shim for the couple of values we use, I had that originally, but decided to change it during review

@bartonjs  thoughts?

I am happy with whatever you both decide